generator/transit_generator.cpp

```diff
@@ -321,7 +245,196 @@ void BuildTransit(string const & mwmDir, string const & countryId,
   w.Seek(startOffset);
   numberSerializer(header);
   w.Seek(endOffset);
+
   LOG(LINFO, (TRANSIT_FILE_TAG, "section is ready. Header:", header));
 }
+
+void GraphData::AppendTo(GraphData const & rhs)
+{
+  ::Append(rhs.m_stops, m_stops);
+  ::Append(rhs.m_gates, m_gates);
+  ::Append(rhs.m_edges, m_edges);
+  ::Append(rhs.m_transfers, m_transfers);
+  ::Append(rhs.m_lines, m_lines);
+  ::Append(rhs.m_shapes, m_shapes);
+  ::Append(rhs.m_networks, m_networks);
+}
+
+void GraphData::Clear()
+{
+  ClearVisitor const v;
+  Visit(v);
+}
+
+bool GraphData::IsValid() const
+{
+  if (!IsSorted() || !IsUnique())
+    return false;
+
+  IsValidVisitor v;
+  Visit(v);
+  return v.IsValid();
+}
+
+void GraphData::Sort()
+{
+  SortVisitor const v;
+  Visit(v);
+}
+
+void GraphData::CalculateEdgeWeights()
+{
+  CHECK(is_sorted(m_stops.cbegin(), m_stops.cend()), ());
```

Comment:
Isn't CHECK too heavy here?
Also, if this is going to be the only place where FindStopById is used (and it is now),
please remove ASSERT(is_sorted) from FindStopById.

Reply:
Yes. Let's remove it from here. Anyway it's checked in ProcessGraph() now. After CalculateEdgeWeights() calls.

Reply:
Done